src/solvers/dgsem_tree/indicators_1d.jl

```diff
@@ -69,8 +69,17 @@ function (indicator_hg::IndicatorHennemannGassner)(u, mesh::Union{TreeMesh{1}, S
     end
 
     # Calculate energy in higher modes
-    energy = max((total_energy - total_energy_clip1) / total_energy,
-                 (total_energy_clip1 - total_energy_clip2) / total_energy_clip1)
+    if !(iszero(total_energy))
+      energy_frac_1 = (total_energy - total_energy_clip1) / total_energy
+    else
+      energy_frac_1 = zero(total_energy)
+    end
+    if !(iszero(total_energy_clip1))
+      energy_frac_2 = (total_energy_clip1 - total_energy_clip2) / total_energy_clip1
+    else
+      energy_frac_2 = zero(total_energy_clip1)
+    end
+    energy = max(energy_frac_1, energy_frac_2)
```

Comment:
Does this produce type stable code?

Reply:
As long as we do everything in a sane way on our side - yes. It will be type stable, e.g., for `Float64` or ForwardDiff Duals

Reply:
OK, thanks /r